roles/rhn-subscription/tasks/osp_repos.yml

```diff
@@ -0,0 +1,15 @@
+---
+
+- name: discover OSP pool subscription status
+  shell: subscription-manager list --consumed | grep {{ redhat.osp.pool_id }}
+  register: pool_results
+  failed_when: false
+
+- name: add OSP pool
+  command: subscription-manager attach --pool={{ redhat.osp.pool_id }}
+  when: pool_results|failed
+
+- name: add OSP repositories
+  command: subscription-manager repos --enable={{ item }}
+  with_items:
+  - "{{ redhat.osp.repos }}"
```

Comment:
since repos are already an array you can just do:

with_items: "{{ redhat.osd.repos }}"